core/common/src/main/java/alluxio/util/ShellUtils.java

```diff
@@ -234,13 +173,71 @@ public String toString() {
   }
 
   /**
-   * Static method to execute a shell command.
+   * Static method to execute a shell command. The StdErr is not returned.
+   * If execution failed
    *
    * @param cmd shell command to execute
    * @return the output of the executed command
+   * @throws IOException in various situations:
+   *  1. {@link ExitCodeException} when exit code is non-zero
+   *  2. when the executable is not valid, i.e. running ls in Windows
+   *  3. execution interrupted
+   *  4. other normal reasons for IOException
    */
   public static String execCommand(String... cmd) throws IOException {
-    return new Command(cmd).run();
+    return new ShellCommand(cmd).run();
+  }
+
+  /**
+   * Static method to execute a shell command and tolerate non-zero exit code.
+   * Preserves exit code. Stderr is redirected to stdout.
+   *
+   * @param cmd shell command to execute
+   * @return the output of the executed command
+   * @throws IOException in various situations:
+   *  1. when the executable is not valid, i.e. running ls in Windows
+   *  2. execution interrupted
+   *  3. other normal reasons for IOException
+   */
+  public static CommandReturn execCommandTolerateFailure(String... cmd) throws IOException {
```

Comment:
rename these to "*execWithOutput`

Reply:
Thanks for the catch. Turned out I missed a few `TolerateFailure` too.